Mage.Sets/src/mage/cards/y/YennaRedtoothRegent.java

```diff
@@ -0,0 +1,131 @@
+    }
+
+    public YennaRedtoothRegent(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{G}{W}");
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.ELF);
+        this.subtype.add(SubType.NOBLE);
+        this.power = new MageInt(4);
+        this.toughness = new MageInt(4);
+
+        // {2}, {T}: Choose target enchantment you control that doesn't have the same name as another permanent you control. Create a token that's a copy of it, except it isn't legendary. If the token is an Aura, untap Yenna, Redtooth Regent, then scry 2. Activate only as a sorcery.
+        Ability ability = new ActivateAsSorceryActivatedAbility(
+                new YennaRedtoothRegentEffect(),
+                new GenericManaCost(2)
+        );
+        ability.addCost(new TapSourceCost());
+        ability.addTarget(new TargetControlledPermanent(filter));
+        this.addAbility(ability);
+    }
+
+    private YennaRedtoothRegent(final YennaRedtoothRegent card) {
+        super(card);
+    }
+
+    @Override
+    public YennaRedtoothRegent copy() {
+        return new YennaRedtoothRegent(this);
+    }
+}
+
+enum YennaRedtoothRegentPredicate implements ObjectSourcePlayerPredicate<Permanent> {
+    instance;
+
+    @Override
+    public boolean apply(ObjectSourcePlayer<Permanent> input, Game game) {
+        String name = input.getObject().getName();
+        if (name == null || name.isEmpty()) {
+            return true;
+        }
+
+        FilterControlledPermanent filter = new FilterControlledPermanent();
+        filter.add(new NamePredicate(name));
+        return game.getBattlefield().count(filter, input.getPlayerId(), input.getSource(), game) == 1;
+    }
+}
+
+class YennaRedtoothRegentEffect extends OneShotEffect {
+
+    YennaRedtoothRegentEffect() {
+        super(Outcome.PutCreatureInPlay);
+        staticText = "choose target enchantment you control that doesn't have the same name as another "
+                + "permanent you control. Create a token that's a copy of it, except it isn't legendary. "
+                + "If the token is an Aura, untap {this}, then scry 2";
+    }
+
+    private YennaRedtoothRegentEffect(final YennaRedtoothRegentEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public YennaRedtoothRegentEffect copy() {
+        return new YennaRedtoothRegentEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        CreateTokenCopyTargetEffect create = new CreateTokenCopyTargetEffect()
+                .setPermanentModifier((token) -> token.removeSuperType(SuperType.LEGENDARY));
+
+        if (!create.apply(game, source)) {
+            return false;
+        }
+
+        Permanent token = create.getAddedPermanents().get(0);
+        if (token == null || !token.hasSubtype(SubType.AURA, game)) {
+            return true;
+        }
```

Comment:
I would check all the created permanents here just in case multiple different kinds of token were created